index.js

```diff
@@ -2,21 +2,67 @@
 
 const express = require('express');
 const app = express();
+const { Client } = require('pg');
+const bodyParser = require('body-parser');
+app.use(
+  bodyParser.text({
+    type: 'text/html'
+  })
+);
 
-app.set('port', (process.env.PORT || 8080));
+const client = new Client({
+  connectionString: process.env.DATABASE_URL,
+  user: 'olamothe',
+  database: 'qa'
+});
+
+client.connect();
+
+app.set('port', process.env.PORT || 8080);
 app.use('/', express.static(__dirname + '/bin'));
 
+app.get('/qa/:pagename', function(request, response) {
+  client.query(`SELECT content FROM pages WHERE name LIKE $1 LIMIT 1;`, [request.params.pagename], (err, res) => {
+    if (err) {
+      response.status(500).send(err.message);
+    } else if (res.rowCount == 1) {
+      return response.send(res.rows[0].content);
+    } else {
+      return response.sendStatus(404);
+    }
+  });
+});
+
+app.post('/qa/:pagename', function(request, response) {
+  if (request.body && request.headers.authentication == `Bearer ${process.env.UPLOAD_KEY}`) {
+    client.query(`SELECT 1 FROM pages WHERE name LIKE $1;`, [request.params.pagename], (err, res) => {
+      if (err) {
+        response.status(500).send(err.message);
+      } else if (res.rowCount == 1) {
+        client.query(`UPDATE pages SET content=$1 WHERE name LIKE $2;`, [request.body, request.params.pagename], (err, res) => {
+          response.sendStatus(200);
```

Comment:
Shouldn't we also handle errors at this level in case a bad body is sent?

Reply:
Yeah, well it's accepting pretty much anything anyway.

It's an internal QA tool, so it does not have to be completely bulletproof either. Lucille wont be changing those pages frequently.

Reply:
Ok, in that case it's time to 🚢.